core/admin.py

```diff
@@ -12,36 +12,46 @@
 from . import tasks
 
 
-def _text_preview(log_file: FileField):
+def _text_preview(target_file: FileField, mimetype, show_end):
     """
     Return the text of a file if it is short or the last portion of it if it is long.
 
-    :params: log_file A FileField to read text from.
+    params: target_file A FileField to read text from.
+            mimetype Log or Data's mimetype, score result don't have one
+            show_end Determine if showing the first/last portion for preview
     """
+    mimetype_check = mimetype.startswith('text/')
```

Comment:
If we allow `mimetype` to be `None`, this will change to `mimetype_check = mimetype is not None and mimetype.startswith('text/')`

Reply:
We should return right away when the mimetype check fails.  We don't need to open the file or check if it exists.